lib/Sema/SemaExpr.cpp

```diff
@@ -13809,6 +13809,27 @@ static bool isImplicitlyDefinableConstexprFunction(FunctionDecl *Func) {
          (Func->isImplicitlyInstantiable() || (MD && !MD->isUserProvided()));
 }
 
+namespace
+{   // TODO: potentially temporary.
+  inline
+  bool is_hip_functor(const CXXMethodDecl* f)
+  {
+    static constexpr const char prefix[] = "HIP_kernel_functor_name_begin";
+
+    return f->getOverloadedOperator() == OO_Call &&
+    f->getParent()->getName().find(prefix) != StringRef::npos;
+  }
+
+  inline
+  void add_callee_attributes_to_functor(
+    const FunctionDecl* callee, FunctionDecl* functor_call_operator)
+  {
+    functor_call_operator->dropAttrs();
```

Comment:
@scchan pretty much; the functor is quite naked and merely a manufactured wrapper. It only stands in for the actual function, so in this case it gives no extra information (by way of attributes).